File: AlgorithmsCSharp/数据结构/828_模拟栈.cs
namespace _828_模拟栈
{
    class Program
    {
        static void Main()
        {
            int m = Convert.ToInt32(Console.ReadLine());
            int[] stack = new int[100010];
            int top = 0;
            while (m-- > 0)
            {
                string op = Console.ReadLine();
                if (op == "push")
                {
                    int x = Convert.ToInt32(Console.ReadLine());
                    top++;
                    stack[top] = x;
                }
                else if (op == "query")
                {
                    Console.WriteLine(stack[top]);
                }
                else if (op == "pop")
                {
                    top--;
                }
                else if (op == "empty")
                {
                    Console.WriteLine(top > 0 ? "NO" : "YES");
                }
            }
        }
    }
}

File: AlgorithmsCpp/数据结构/828_模拟栈.cpp
# include<iostream>

using namespace std;

const int N = 100010;
int m, st[N], top;

int main() {
	cin >> m;
	while (m--) {
		string s;
		cin >> s;
		if (s == "push") {
			int x;
			cin >> x;
			st[++top] = x;
		}
		else if (s == "query") {
			cout << st[top] << endl;
		}
		else if (s == "pop") {
			top--;
		}
		else {
			if (top) cout << "NO" << endl;
			else cout << "YES" << endl;
		}
	}
	return 0;
}